Release electron app

bonomat opened the pull request:
Draft for now. 
Seems to be working on my fork: https://github.com/bonomat/hermes/actions/runs/3110831685/jobs/5042429562

commit history can be cleaned-up but this is a sneak peek

resolves https://github.com/itchysats/itchysats/issues/2980

bonomat commented on taker-electron/forge.config.js:
```diff
@@ -0,0 +1,33 @@
+                // todo: change to itchysats/itchysats
+                "repository": {
+                    "owner": "bonomat",
+                    "name": "hermes",
```
Fix me!
```suggestion
                    "owner": "itchysats",
                    "name": "itchysats",
```

bonomat commented on taker-electron/forge.config.js:
```diff
@@ -0,0 +1,33 @@
+        {
+            "name": "@electron-forge/publisher-github",
+            "config": {
+                // todo: change to itchysats/itchysats
```
```suggestion
```

bonomat committed:
chore(ci): bump node version

Signed-off-by: Philipp Hoenisch <philipp@coblox.tech>

luckysori reviewed (commented):
I'm confused about most of the changes in `taker-electron`. They seem unrelated to a PR that should mostly affect our GitHub Actions. Can we remove some of those changes or extract hem into a separate commit?

klochowicz commented on taker-electron/package.json:
```diff
@@ -1,71 +1,146 @@
-  "version": "0.1.0",
-  "description": "",
-  "main": "dist/main.js",
+  "version": "0.6.2",
```
can this be bumped automatically? (if not, we should probably add it to our release checklist)

bonomat replied:
This one is not that critcal but can be bumped automatically as well.

klochowicz commented on dprint.json:
```diff
@@ -18,7 +18,10 @@
     "maker-frontend/dist",
     "taker-frontend/dist",
```
```suggestion
```

the new additions make it redundant

klochowicz commented on dprint.json:
```diff
@@ -18,7 +18,10 @@
+    "rocket-cookie-auth/example-http-cookie-auth.sql",
+    "**/dist",
+    "**/build",
+    "taker-electron/testnet"
```
```suggestion
    "**/testnet"
```

this will make it easier to ignore all the testnet dirs. 
BTW: what dprint files are in the testnet dir of taker-electron that you wanted to exclude specifically?

bonomat replied:
Also does not exist anymore :D

klochowicz commented on taker-electron/.editorconfig:
```diff
@@ -0,0 +1,12 @@
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
why not trim trailing whitespaces in md files? trailing whitespaces are universally bad 😝

bonomat replied:
outdated file from the boilerplate. Thanks. I'll delete it.

klochowicz commented on dprint.json:
```diff
@@ -18,7 +18,10 @@
     "taker-frontend/dist",
     "**/node_modules",
-    "rocket-cookie-auth/example-http-cookie-auth.sql"
+    "rocket-cookie-auth/example-http-cookie-auth.sql",
```
why make this special snowflake? is there a reason why it shouldn't be formatted consistently too? (also: it's completely unrelated to the PR)

bonomat replied:
Actually this file does not exist anymore...

klochowicz replied:
then we can remove the special case, problem solved eitherway :-)

klochowicz commented on taker-electron/src/main/menu.ts:
```diff
@@ -0,0 +1,257 @@
+            label: "Electron",
+            submenu: [
+                {
+                    label: "About ElectronReact",
```
disclaimer: I haven't tested this, so it might be a wrong suggestion 🙃 especially if there's some magic code that replaces that string later on with the name of the app.

isn't this how the menu will be showed and shouldn't we tailor the following? 

```suggestion
            label: "ItchySats",
            submenu: [
                {
                    label: "About ItchySats",
```

bonomat replied:
Good find

klochowicz commented on taker-electron/src/main/menu.ts:
```diff
@@ -0,0 +1,257 @@
+                { label: "Services", submenu: [] },
+                { type: "separator" },
+                {
+                    label: "Hide ElectronReact",
```
(same as above)

```suggestion
                    label: "Hide ItchySats",
```

klochowicz commented on taker-electron/src/main/menu.ts:
```diff
@@ -0,0 +1,257 @@
+                        },
+                    },
+                    {
+                        label: "FAQ",
```
super cool to have the FAQ link there!!! 💯

other ideas for additional items:
-  `Join  Community` with the telegram group link
- `Ask for Help` with the link to github discussions

bonomat committed:
chore(electron): use electron-builder instead of electron-forge

- build release binary for windows x64
- build release binary for macos x64 and arm64
- signs macos binaries using signing keys

Signed-off-by: Philipp Hoenisch <philipp@coblox.tech>

bonomat committed:
feature(electron): add release taker-electron pipeline

It expects `APPLE_ID` and `APPLE_ID_PASS` to be present or the macos binaries won't be signed.

Signed-off-by: Philipp Hoenisch <philipp@coblox.tech>

bonomat committed:
chore(dprint): this file does not exist anymore

Signed-off-by: Philipp Hoenisch <philipp@coblox.tech>

luckysori reviewed (approved):
LGTM although I'm still not sure about all those files in the first commit. Are all those changes to `taker-electron` intentional?

klochowicz commented on .github/workflows/draft-new-release.yml:
```diff
@@ -40,6 +40,23 @@ jobs:
           version: ${{ github.event.inputs.version }}
           manifest: daemon/Cargo.toml
 
+      - name: Bump version for taker-electron build file
+        uses: jaywcjlove/github-action-package@main
+        with:
+          path: taker-electron/release/app/package.json
+          data: |
+            {
+              "version": "${{github.event.inputs.version}}"
+            }
+      - name: Bump version for taker-electron package file
+        uses: jaywcjlove/github-action-package@main
+        with:
+          path: taker-electron/package.json
+          data: |
+            {
+              "version": "${{github.event.inputs.version}}"
+            }
+
```
👏

klochowicz reviewed (approved):
great job!!! 

thanks for signing the binary, I can confirm it's much easier to install the new binary (and it works!) ;-)
